### PR TITLE
mutable item stacks and random stuff

### PR DESCRIPTION
**Pull request by friendlyhj**
The PR adds 1.16 MutableItemStack to 1.12. I have checked and changed methods to ensure the origin IItemStack still immutable.

### CODE REVIEW

**Comment by democat3457** [comment]
doesn't contenttweaker have a mutableitemstack

**Comment by friendlyhj** [comment]
Of course I know, but the CoT mutable itemstack only can be used in CoT scripts, not in CrT events.

File: CraftTweaker2-API/src/main/java/crafttweaker/api/util/IRandom.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package crafttweaker.api.util;
+
+import crafttweaker.annotations.ZenRegister;
+import stanhebben.zenscript.annotations.ZenClass;
+
+@ZenClass("crafttweaker.util.IRandom")
+@ZenRegister
+public interface IRandom {
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/util/IRandom.java:9]: please add a `nextFloat`

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

**Comment by jaredlll08** [comment]
I believe this looks good, but please can you test it with some big 1.12 modpacks to make sure the mutable changes don't cause any issues

File: CraftTweaker2-API/src/main/java/crafttweaker/api/util/IRandom.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package crafttweaker.api.util;
+
+import crafttweaker.annotations.ZenRegister;
+import stanhebben.zenscript.annotations.ZenClass;
+import stanhebben.zenscript.annotations.ZenMethod;
+
+@ZenClass("crafttweaker.util.IRandom")
+@ZenRegister
+public interface IRandom {
```
Comment (democat3457) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/util/IRandom.java:9]: can you also add the methods that are found in MathHelper that take a rand as a parameter - for example, the nextX with min and max bound methods

Comment (democat3457) [reply CraftTweaker2-API/src/main/java/crafttweaker/api/util/IRandom.java:9]: (these are the `getInt`, `getRandomUUID`, `nextFloat`, and `nextDouble` methods, specifically)

**Comment by friendlyhj** [comment]
> I believe this looks good, but please can you test it with some big 1.12 modpacks to make sure the mutable changes don't cause any issues

Tested, I believe these changes don't

**Pull request merged**